packages/lambda-shared/src/aws/tile.metadata.table.ts

```diff
@@ -111,29 +120,30 @@ export class TileMetadataTable {
      * @param Keys Imagery ids (already prefixed `im_${key}`)
      */
     private async fetchImagery(keys: string[]): Promise<TileMetadataImageryRecord[]> {
-        // TODO run multiple requests
-        if (keys.length > 100) throw new Error('BatchGet only allows 100 keys');
-
-        const Keys = keys.map(toId);
-        const items = await this.dynamo
-            .batchGetItem({
-                RequestItems: {
-                    [Const.TileMetadata.TableName]: { Keys },
-                },
-            })
-            .promise();
-
-        const metadataItems = items.Responses?.[Const.TileMetadata.TableName];
-        if (metadataItems == null) throw new Error('Failed to fetch tile metadata');
-
-        // TODO diff the arrays see what is missing?
-        if (metadataItems.length < Keys.length) throw new Error('Missing fetched items');
-
-        const results = metadataItems.map((c) => DynamoDB.Converter.unmarshall(c)) as TileMetadataImageryRecord[];
-
-        for (const result of results) this.imagery.set(result.id, result);
+        let mappedKeys = keys.map(toId);
+        let results: TileMetadataImageryRecord[] | null = null;
+
+        while (mappedKeys.length != 0) {
+            const Keys = mappedKeys.length > 100 ? mappedKeys.slice(0, 100) : mappedKeys;
+            mappedKeys = mappedKeys.length > 100 ? mappedKeys.slice(100) : [];
+
+            const items = await this.dynamo
+                .batchGetItem({
+                    RequestItems: {
+                        [Const.TileMetadata.TableName]: { Keys },
+                    },
+                })
+                .promise();
+
+            const metadataItems = items.Responses?.[Const.TileMetadata.TableName];
+            if (metadataItems == null) throw new Error('Failed to fetch tile metadata');
+
+            const res1 = metadataItems.map((c) => DynamoDB.Converter.unmarshall(c)) as TileMetadataImageryRecord[];
```

Comment:
we are now looping over this about three times, maybe we should return a map of imagery

then we can just 
```
for (const item of metadataItems) {
   const i = Convert(item);
   output.set(i.id, i)
   this.imagery.set(i.id, i)
}
```

Reply:
I'll do some more work on this